File: deletionAtIndex.c
// Program to implement Data structure array deletion operation at specific index
#include <stdio.h>
#include <conio.h>

void display(int arr[], int n)
    // Array Traversal
{
    //Looping through array elements and printing them
    for (int i=0; i<n; i++)
    {
        printf("%d\n",arr[i]);
    }
}

// Function for deletion of array element at specific index
void indexDeletion(int arr[], int size, int index)
{
// code for deletion operation
for (int i=index; i<size-1; i++)
{
    arr[i] = arr[i+1];
}
printf("Element at index %d has been deleted successfully\n", index);
}

int main()
{
    int arr[100] = {7,8,12,27,88};
    int size=5, index=2;
    display(arr, size);
    indexDeletion(arr, size, index);
    size -= 1;
    printf("New array after deletion operation\n");
    display(arr, size);
    return 0;
}

File: Data Structure Array Programs in C, JS and Ruby/deletionAtIndex.js
// Program to demonstrate Data Structure array delete operation at a specific index.
function display(arr, size) {
    // Array Traversal
  // Loop through array elements and print them
    for (let i = 0; i < size; i++) {
        console.log(`The value at index ${i} is ${arr[i]}`);
    }
}

function indexDeletion(arr, size, index) {
    // Check if the array is empty
    if (size === 0) {
        return -1;
    }
    // Deleting element at specific index
    for (let i = index; i < size - 1; i++) {
        arr[i] = arr[i + 1];
    }
    console.log(`Element at index ${index} deleted successfully!`);
    return 1;
}

function main() {
    let arr = [7, 8, 12, 27, 88];
    let size = 5;
    const index = 1;
    display(arr, size);
    // Deleting element at index
    indexDeletion(arr, size, index);
    // Decrease the size of the array after deletion
    size -= 1;
    console.log("After Deletion");
    // Print the array after deletion operation
    display(arr, size);
}

main();